feat(redis adapter): create redis adapter as secondary database

Marius-brt opened the pull request:
This PR introduces Redis adapter as a Secondary Database. It allow to add Redis in one line of code instead to have to do manual implementation. It use ioredis under the hood.

### Changes

- Create Redis Adapter by implementing the `SecondaryStorage` interface
- Create tests for the adapter
- Add Redis in the `docker-compose.yml`
- Add a page in the documentation

ping-maxwell commented:
Interesting idea. Never thought about secondaryDB adapter🤔

Pukimaa commented on docs/content/docs/adapters/redis.mdx:
```diff
@@ -0,0 +1,33 @@
+
+Redis is an open-source, in-memory data structure store that can be used as a database, cache, and message broker.
+In Better Auth, Redis can be used as a session store to manage user sessions and authentication tokens.
+Read more [here](https://redis.io/).
```
For accessibility reasons it's best to avoid linked text like "here". I recommend adding something like "about Redis here".

```suggestion
Read more [about Redis here](https://redis.io/).
```

Pukimaa commented on docs/content/docs/adapters/redis.mdx:
```diff
@@ -0,0 +1,33 @@
+export const auth = betterAuth({
+    ...,
+    secondaryStorage: redisAdapter({
+        connectionString: "redis://localhost:6379",
```
I'd add the same example you have in the redis adapter type docs as well.
```suggestion
        connectionString: "redis://user:password@localhost:6379",
```

Pukimaa reviewed (changes requested):
Aside from some minor documentation improvements that could be made, I really like this contribution!
Although maybe someone else should also look over the actual code, because I'm not too proficient with Redis or secondary storage in Better Auth :)

ping-maxwell commented:
@Marius-brt Are you able to rename the redisAdapter to something like redisSecondaryAdapter?
I'm currently developing a redisAdapter - for primary DB and this would be conflicting names 😅
If not it's fine, the dev can always map/rename the imports, but would be nice if this could be renamed! ❤️

Marius-brt commented:
@ping-maxwell I updated the location of the adapter to a new folder called `secondary-adapters`. So the import is now `import { redisSecondaryAdapter } from "better-auth/secondary-adapters/redis";`. 
I also updated the doc. I hope this folder structure is good for you 👍

ping-maxwell commented:
Oh this is great! Thanks so much, appreciate the cooperation ❤️

himself65 commented:
Thank you for your contribution! 🙏

The `main` branch is now protected and we're no longer accepting PRs directly to it. Could you please reopen this PR targeting the `canary` branch instead?

To update your PR:
```bash
git checkout your-branch
git push origin your-branch:canary
```

Or you can create a new PR targeting `canary`. We appreciate your understanding!
